core/libp2p/event/emitter.hpp

```diff
@@ -102,7 +104,11 @@ namespace libp2p::event {
      */
     template <typename Event>
     Subscription on(const std::function<void(const Event &)> &handler) {
-      return Base::template getSignal<Event>().connect(handler);
+      // TODO(@warchant): check that Event exists in Events
+      // possible solution (fails compilation):
+      //      static_assert((std::is_same_v<std::decay_t<Event>, Events> ||
+      //      ...));
+      return Subscription(Base::template getSignal<Event>().connect(handler));
```

Comment:
This TODO is not needed. I have just re-tested, if you call an event, which is not there, it fails the compilation, because the recursive template cannot find it. You can re-check my statement in the EventEmitterTest, trying to call an event, not registered in it